### ISSUE TITLE
Move opinionated rules from 'eslint-recommended' to 'recommended'?

### ISSUE THREAD

**Issue by borekb** [typescript-eslint/typescript-eslint#592]
While looking at #591, I encountered something that I wanted to discuss separately.

Intuitively, I would think that a typical config could look like this:

```js
{
  extends: [
    // start from this
    'eslint:recommended',

    // disable rules that are incompatible with TS or better handled by it
    '@typescript-eslint/eslint-recommended',

    // turn on TS-specific recommended rules
    '@typescript-eslint/recommended'
  ]
```

However, `@typescript-eslint/eslint-recommended` [already contains some opinionated rules](https://github.com/typescript-eslint/typescript-eslint/blob/b5e95edbb811ba239d2856e9caee21e802fbd8fb/packages/eslint-plugin/src/configs/eslint-recommended.ts#L43-L47) so adding it to our config actually yielded some new errors, which I wouldn't have expected.

I think I'd prefer if `eslint-recommended` was just a compatibility config while all the additional, opinionated and possibly stricter rules were in `recommended`. What do you think?

**Comment by bradzacher** [comment]
I'm okay with this!
But we'd have to wait for a major bump, as it's technically a breaking change.

**Comment by CvX** [comment]
I like this proposal since, especially since currently adding "@typescript-eslint/eslint-recommended" to a config enables these rules using the `overrides`, which results in an unexpected behavior when you try to disable any of them, e.g.

```js
{
  rules: {
    'prefer-const': 'off',
  },
}
```

…doesn't work. Instead, you have to also use an override:

```js
{
  overrides: [
    {
      files: ['*.ts', '*.tsx'],
      rules: { 'prefer-const': 'off' },
    },
  ],
}
```

**Comment by bradzacher** [comment]
Off topic, but out of curiosity @CvX - why don't you want to use `prefer-const`?

**Comment by CvX** [comment]
Two reasons:

1. I'm replacing tslint with eslint/typescript-eslint, so I'm getting a lot of new warnings/issues. What I like to do, is to disable most of them, and then incrementally deal with them, enabling and fixing them one by one.
2. `prefer-const` in particular isn't that useful, since I care more about immutability rather than variable reassignment. The distinction between `let` and `const` often feels unnecessary.
